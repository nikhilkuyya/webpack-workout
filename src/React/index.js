import React from "react"
import ReactDOM from "react-dom"

const startReactApp = node => {
  const App = require("./App").default
  ReactDOM.render(React.createElement(App), node)
}

const unmountReactApp = node => {
  ReactDOM.unmountComponentAtNode(node)
}

module.hot.accept(startReactApp)

export { startReactApp, unmountReactApp }
