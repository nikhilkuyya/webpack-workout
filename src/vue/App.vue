<template >
    <div :id='rootId'>{{framework}}</div>
</template>

<script>
export default {
  data() {
    return {
      rootId: "app",
      framework: "Vue"
    }
  }
}
</script>