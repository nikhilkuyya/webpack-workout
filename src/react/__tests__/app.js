import { render } from "react-testing-library"
import React from "react"
import App from "../app"

test("react app renders", () => {
  const { container } = render(<App />)
  expect(container).toMatchSnapshot()
})
