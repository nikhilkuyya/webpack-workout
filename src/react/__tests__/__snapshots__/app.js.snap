// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react app renders 1`] = `
<div>
  <div
    class="main"
  >
    React
  </div>
  0
</div>
`;
